inspire/modules/citations/tasks.py

```diff
@@ -54,6 +55,41 @@ def update_records_citations(new_citations):
         citees.clear()
 
 
+@before_record_insert.connect
+def catch_citations_insert(sender, *args, **kwargs):
+    a = []
+    if 'references' in sender:
+        references = sender['references']
+        for reference in references:
+            recid = reference.get('recid')
+            if recid is not None:
+                a.append(recid)
+
+
+@before_record_update.connect
+def catch_citations_update(sender, *args, **kwargs):
+    a = []
+    if 'references' in sender:
+        references = sender['references']
+        for reference in references:
+            recid = reference.get('recid')
+            if recid is not None:
+                a.append(recid)
+    b = []
+    if 'control_number' in sender:
+        from invenio_ext.es import es
+        recid = sender['control_number']
+        rec = es.get(index='hep', id=recid)
```

Comment:
This might throw an exception when a non-hep record is modified as they are stored in a different index